HPXMLtoOpenStudio/resources/battery.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+class Battery
+  def self.apply(model, battery)
+    obj_name = battery.id
+
+    default_values = Battery.get_battery_default_values()
+
+    voltage = battery.voltage # V
+    capacity = get_kWh_from_Ah(battery.capacity, voltage) # kWh
+
+    return if capacity <= 0 || voltage <= 0
+
+    is_outside = (battery.location == HPXML::LocationOutside) || (battery.location == HPXML::LocationExteriorWall) || (battery.location == HPXML::LocationRoofDeck)
+    if not is_outside
+      frac_sens = 1.0
+    else # Internal gains outside unit
+      frac_sens = 0.0
+    end
+
+    # The following calculations are from Rohit C.
+    number_of_cells_in_series = Integer((voltage / 3.6).round)
+    number_of_strings_in_parallel = Integer(((capacity * 1000.0) / (voltage * 3.2)).round)
+    battery_mass = (capacity / default_values[:capacity]) * 99.0 # kg
+    battery_surface_area = 0.306 * (capacity**(2.0 / 3.0)) # m^2
+
+    elcds = model.getElectricLoadCenterDistributions
+    if elcds.empty?
+      elcd = OpenStudio::Model::ElectricLoadCenterDistribution.new(model)
+      elcd.setName('Battery elec load center dist')
+      elcd.setElectricalBussType('AlternatingCurrentWithStorage')
+    else
+      elcd = elcds[0]
+
+      if elcd.inverter.is_initialized
+        elcd.setElectricalBussType('DirectCurrentWithInverterDCStorage')
+      end
+    end
+
+    # elcd.setGeneratorOperationSchemeType('TrackElectrical')
+    # elcd.setDemandLimitSchemePurchasedElectricDemandLimit(0)
+    # elcd.setStorageOperationScheme('TrackFacilityElectricDemandStoreExcessOnSite')
+    # elcd.setMaximumStorageStateofChargeFraction(0.95)
+    # elcd.setMinimumStorageStateofChargeFraction(0.20)
+
+    elcs = OpenStudio::Model::ElectricLoadCenterStorageLiIonNMCBattery.new(model, number_of_cells_in_series, number_of_strings_in_parallel, battery_mass, battery_surface_area)
+    elcs.setName("#{obj_name} li ion")
+    unless is_outside
+      space = battery.additional_properties.space
+      thermal_zone = space.thermalZone.get
+      elcs.setThermalZone(thermal_zone)
+    end
```

Comment:
I was under the impression that the location of the battery matters because of A) where the heat losses end up and B) the temperature that the battery sees. For a battery located outside, it's unclear to me how B is addressed.

Reply:
@nmerket will verify that the battery sees outdoor temperatures when no thermal zone is specified.

Reply:
Confirmed by @nmerket. https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/ElectricPowerServiceManager.cc#L4303